src/coreclr/jit/morph.cpp

```diff
@@ -3292,7 +3292,16 @@ void Compiler::fgInitArgInfo(GenTreeCall* call)
 
         if (isStructArg)
         {
-            assert(argx == args->GetNode());
+#if defined TARGET_WASM
+            // LLVM/Wasm wraps all args in a PUTARG when lowering, but not in morphing so have to handle both ways
+            // .. or exclude the assert for LLVM 
+            if(args->GetNode()->OperIs(GT_PUTARG_TYPE))
+            {
+                assert(argx == args->GetNode()->AsPutArgType()->gtGetOp1());
+            }
+            else
+#endif
+                assert(argx == args->GetNode());
```

Comment:
Honestly, doesn't seem terribly valuable to have this large `#ifdef` structure, can be just:
```cpp
#ifndef TARGET_WASM // We don't update the node in the arg info for LLVM.
assert(argx == args->GetNode());
#endif
```
We should target getting rid of the codegen's dependency on arg info and use the `PUTARG_TYPE`s to relay the relevant information.

Reply:
I think this is going to be easy so I'll just try and do it, removing these `#ifs`

Reply:
`morph.cpp` no longer appears in the list of changed files.